### PR TITLE
Draw card fix for human and code cleanup

### PR DESCRIPTION
**Pull request by krithigasr**
Contains the following code changes
* Bug Fix for draw card for human player
* Code clean up
* Adding human input rules

### CODE REVIEW

File: src/cards_management.c [hunk 1]
```diff
@@ -87,10 +88,10 @@ int initialize_cards(void)
 }
 
 /**
- * @brief Deals each player 5 cards at the start of the game setup
+ * @brief Deals each player 5 cards during the game initialization
```
Comment (MiaYWei) [inline src/cards_management.c:91]: Please update the corresponding description in the header file as well.

File: src/human_player_module.c [hunk 2]
```diff
@@ -104,18 +104,26 @@ ret_type_e record_human_input(void)
     }
     else if (user_input[0] == 'r' || user_input[0] == 'R')
```
Comment (MiaYWei) [inline src/human_player_module.c]: Please change "r" to other character, since "r" is also for color RED.

File: src/cards_management.c (hunk 1 above)
Comment (krithigasr) [reply src/cards_management.c:91]: Done

File: src/human_player_module.c (hunk 2 above)
Comment (krithigasr) [reply src/human_player_module.c]: Done

**Review by HongBo713 — APPROVED** [review]

**Review by MiaYWei — APPROVED** [review]

**Review by aditibhardwaj98 — APPROVED** [review]

**Pull request merged**